chia/server/address_manager.py

```diff
@@ -300,7 +300,7 @@ def clear_new_(self, bucket: int, pos: int) -> None:
     def mark_good_(self, addr: PeerInfo, test_before_evict: bool, timestamp: int) -> None:
         self.last_good = timestamp
         (info, node_id) = self.find_(addr)
-        if not addr.is_valid(self.allow_private_subnets):
+        if addr.ip.is_private and not self.allow_private_subnets:
```

Comment:
it's not obvious that the new condition is the same as the old. Is being in a private subnet while private subnets are disallowed the only way an IP can be invalid in?
The left side suggests the IP address can be invalid in more ways

Reply:
It's just, before we validated the host in [PeerInfo.is_valid](https://github.com/Chia-Network/chia-blockchain/blob/99abc2d66f0136bf6d0420a9c3163664987a04ee/chia/types/peer_info.py#L16), see below. But with this PR we store valid IP addresses in `PeerInfo` so `is_valid` is not longer needed and gets dropped. The condition `addr.ip.is_private and not self.allow_private_subnets` is just the private subnet check which was nested into `PeerInfo.is_valid` for some reason.